handler_ext_test.go

```diff
@@ -62,4 +67,32 @@ func TestHandler_ServeHTTP(t *testing.T) {
 			"application/proto",
 		}, ", "))
 	})
+
+	t.Run("unsupported_content_encoding", func(t *testing.T) {
+		t.Parallel()
+		req, err := http.NewRequest(http.MethodPost, server.URL+pingProcedure, strings.NewReader("{}"))
+		assert.Nil(t, err)
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Content-Encoding", "invalid")
+		resp, err := client.Do(req)
+		assert.Nil(t, err)
+		defer resp.Body.Close()
+		type errorMessage struct {
+			Code, Message string
+		}
+		var message errorMessage
+		err = json.NewDecoder(resp.Body).Decode(&message)
+		assert.Nil(t, err)
+		assert.Equal(t, message.Message, `unknown compression "invalid": supported encodings are gzip`)
+		assert.Equal(t, message.Code, connect.CodeUnimplemented.String())
+		assert.Equal(t, resp.StatusCode, http.StatusNotFound)
```

Comment:
Nit, closer to where response is returned. Reading this top-down I was curious what the status code was and didn't expect it to be at the end.